### PR TITLE
Update Owner Ptr and other smart ptrs with const and noexcept

### PR DESCRIPTION
**Pull request by jparisu**
Signed-off-by: jparisu <javierparis@eprosima.com>

### CODE REVIEW

File: cpp_utils/include/cpp_utils/types/Singleton.hpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file Atomicable.hpp
+ *
+ * This file contains class Atomicable definition.
+ */
```
Comment (juanlofer-eprosima) [inline cpp_utils/include/cpp_utils/types/Singleton.hpp:19]: Update

File: cpp_utils/include/cpp_utils/memory/impl/LesseePtr.hpp [hunk 2]
```diff
@@ -60,26 +60,25 @@ class LesseePtr final
...
      * It could not contain any data.
      */
-    LesseePtr();
+    LesseePtr() noexcept = default;
 
     /**
      * @brief Destroy the Lessee Ptr object
      *
      * It does not destroy the data (it has not the ownership).
+     * Mutex of Internal reference is handled by GuardPtrs.
```
Comment (juanlofer-eprosima) [inline cpp_utils/include/cpp_utils/memory/impl/LesseePtr.hpp:69]:
```suggestion
     * Mutex of Internal reference is handled by GuardedPtrs.
```

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

File: cpp_utils/include/cpp_utils/types/Singleton.hpp (hunk 1 above)
Comment (jparisu) [reply cpp_utils/include/cpp_utils/types/Singleton.hpp:19]: Wrong file

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

**Pull request merged**